### DISCUSSION TITLE
Question: Projector setup

### DISCUSSION THREAD

**Discussion by TerraSkye** [looplab/eventhorizon#359] [Q&A] [answered]
Hi was looking trough the examples and saw stated the folowing. and i wondered why it is that way :
https://github.com/looplab/eventhorizon/blob/68284b69753cd985ee0e62f7cf196b4ffc69822b/examples/guestlist/domains/guestlist/projectors.go#L184
in here (guestlist projector ) we pass on the repo, and do the saving within the projector
https://github.com/looplab/eventhorizon/blob/68284b69753cd985ee0e62f7cf196b4ffc69822b/examples/guestlist/domains/guestlist/projectors.go#L99
in the invitation projector itself. we return the model, and do the saving outside the projecter.
why are these handled differently? and what are the determing factors to use either one.

**Comment by maxekman** [comment, marked as answer]
In general when there is a one-to-one relationship between an aggregate ID and projected entity ID a projector should be used.

The GuestListProjector is an example of a different kind of projector which is not one-to-one with the event IDs, it uses a fixed ID (for the duration of the test/run) in this case which is used as a statistic collector.

**Comment by TerraSkye** [comment]
thank you for clarifying, it makes sense now!
